internal/services/notification/notification_test.go

```diff
@@ -174,6 +177,94 @@ func TestRunWebhookDelivery(t *testing.T) {
 	})
 }
 
+func TestLastRunEventSequence(t *testing.T) {
+	t.Parallel()
+
+	dir := t.TempDir()
+	log := testutil.NewLogger(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	ns := setupNotificationService(ctx, t, log, dir)
+	ns.c.WebhookURL = webhookURL
+
+	t.Logf("starting ns")
+
+	runEventsSender := setupRunEventsSender(ctx, t)
+	defer runEventsSender.stop()
+
+	t.Logf("starting run events service")
+
+	ns.runserviceClient = rsclient.NewClient(runEventsSender.exposedURL, "")
+
+	lastRunEventSequence := getLastRunEventSequenceValue(t, ctx, ns)
+	if lastRunEventSequence != 0 {
+		t.Fatalf("expected lastRunEventSequence %d, got: %d", 0, lastRunEventSequence)
+	}
+
+	// test runEventsHandler start from sequence 0
+	runEventsSender.runEvents.addRunEvent(generateRunEvent(1, rstypes.RunPhaseChanged))
+	runEventsSender.runEvents.addRunEvent(generateRunEvent(2, "testrunevents"))
+	expectedLastRunEventSequenceValue := uint64(2)
+
+	err := ns.runEventsHandler(ctx)
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	lastRunEventSequenceValue := getLastRunEventSequenceValue(t, ctx, ns)
+	if lastRunEventSequenceValue != expectedLastRunEventSequenceValue {
+		t.Fatalf("expected %d last run event got: %d", expectedLastRunEventSequenceValue, lastRunEventSequenceValue)
+	}
+
+	// test runEventsHandler start from last RunEvent processed
+	runEventsSender.runEvents.addRunEvent(generateRunEvent(3, rstypes.RunPhaseChanged))
+	runEventsSender.runEvents.addRunEvent(generateRunEvent(4, "testrunevents"))
+	expectedLastRunEventSequenceValue = 4
+
+	err = ns.runEventsHandler(ctx)
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	lastRunEventSequenceValue = getLastRunEventSequenceValue(t, ctx, ns)
+	if lastRunEventSequenceValue != expectedLastRunEventSequenceValue {
+		t.Fatalf("expected %d last run event got: %d", expectedLastRunEventSequenceValue, lastRunEventSequenceValue)
+	}
```

Comment:
should also test starting from an already process event and check that it's skipped.

Reply:
Ok. This isn't going to be tested... Will add test later...